### ISSUE TITLE
Exception when loading MAPS MIDI file

### ISSUE THREAD

**Issue by carlthome** [craffel/pretty-midi#112]
[MAPS](http://www.tsi.telecom-paristech.fr/aao/en/2010/07/08/maps-database-a-piano-database-for-multipitch-estimation-and-automatic-transcription-of-music/) is a very popular dataset of piano recordings and MIDI annotations for MIR research. Unfortunately it doesn't seem to be supported by pretty_midi though. Am I missing something obvious?

```py
from pretty_midi import PrettyMIDI
PrettyMIDI('MAPS/ENSTDkAm/MUS/MAPS_MUS-ty_mai_ENSTDkAm.mid')
```
> ValueError: MIDI file has a largest tick of 14325216, it is likely corrupt

Example file for debugging: [MAPS_MUS-ty_mai_ENSTDkAm.mid.zip](https://github.com/craffel/pretty-midi/files/612070/MAPS_MUS-ty_mai_ENSTDkAm.mid.zip)

**Comment by craffel** [comment]
If that value error is being raised, that's likely a bug with the MIDI file, not `pretty_midi`.  Having a tick that large means that two events are spaced roughly 8 hours apart.  I don't think there are any reasonable MIDI files for which that is true.  If you can verify that there is not actually a tick that large in the MIDI file, we can investigate.

**Comment by carlthome** [comment]
PPQ is 32767, but is that a bug really? The MIDI files work well in Ardour, can be played with timidity, read with mido, etc. It's an extremely high time resolution, sure.

**Comment by craffel** [comment]
Oh wow, ok.  I mean, I would argue that a PPQ of 32767 is the actual bug.  Yeah, in that case that is just about 3 minutes (which is still a really large spacing between events).  The issue is that `pretty_midi` keeps a cache of tick-to-time conversion, and if ticks like 14325216 are allowed, then the array can get very large (at least 100 megabytes in this case).  The cached tick-to-time array is really a requirement, else a lot of operations would be extremely slow.  The best solution may be to allow the user to override the MAX_TICK setting in a non-hacky way; of course it's totally possible currently like this:

```
In [1]: import pretty_midi

In [2]: pm = pretty_midi.PrettyMIDI()

In [3]: pm.tick_to_time(14325216)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-3-28e6872f7fb3> in <module>()
----> 1 pm.tick_to_time(14325216)

/usr/local/lib/python2.7/dist-packages/pretty_midi-0.2.6-py2.7.egg/pretty_midi/pretty_midi.pyc in tick_to_time(self, tick)
    915         # Check that the tick isn't too big
    916         if tick >= MAX_TICK:
--> 917             raise IndexError('Supplied tick is too large.')
    918         # If we haven't compute the mapping for a tick this large, compute it
    919         if tick >= len(self.__tick_to_time):

IndexError: Supplied tick is too large.

In [4]: pretty_midi.pretty_midi.MAX_TICK = 1e10

In [5]: pm.tick_to_time(14325216)
Out[5]: 32557.30909090909
```

**Comment by carlthome** [comment]
Yeah, it's no fun trying to render a ndarray piano roll with that PPQ. :laughing: 

I think pretty much every song in MAPS could do with a PPQ of almost as low as 96, so perhaps it's best to just leave it up to the user to downsample. Thanks for the hack!

**Comment by JitendraChitre** [comment]
For MAPS_MUS-mendel_op62_5_ENSTDkAm.mid, I am getting an error from pretty_midi, viz. "ValueError: MIDI file has a largest tick of 11731919, it is likely corrupt". This is one of the 30 other files in the MAPS dataset, which pretty midi is falling over with the same ValueError exception.  But there are other 9 which work perfectly fine with pretty_midi for e.g. MAPS_MUS-chpn_op25_e3_ENSTDkAm.mid
Next, I open the above bad file - MAPS_MUS-mendel_op62_5_ENSTDkAm.mid in Windows Media Player and it played fine. Next I opened this same file in Audacity and that worked fine too. So, next I exported this file from Audacity as a midi file and saved it with a different name, MAPS_MUS-mendel_op62_5_ENSTDkAm_new.mid.
Next, I tried to open the new file MAPS_MUS-mendel_op62_5_ENSTDkAm_new.mid using pretty_midi and this time it opened fine. This new file also now opens fine with Audacity as well as Windows media-player.
So, looks like there is something corrupt with the original MAPS_MUS-mendel_op62_5_ENSTDkAm.mid file. Windows-media-player and Audacity are able to overlook this corruption and carry on, but pretty-midi does not cope with the potential value error.
